tests/display/Container.Visual.tests.ts

```diff
@@ -0,0 +1,177 @@
+                expect(mask1.measurable).toBe(true);
+                expect(mask2.includeInBuild).toBe(true);
+                expect(mask2.measurable).toBe(true);
+
+                container1.mask = mask1;
+
+                expect(mask1.includeInBuild).toBe(false);
+                expect(mask1.measurable).toBe(false);
+                expect(mask2.includeInBuild).toBe(true);
+                expect(mask2.measurable).toBe(true);
+
+                container1.mask = mask1;
+
+                expect(mask1.includeInBuild).toBe(true);
+                expect(mask1.measurable).toBe(false);
+                expect(mask2.includeInBuild).toBe(false);
+                expect(mask2.measurable).toBe(true);
+
+                container2.mask = mask1;
+
+                expect(mask1.includeInBuild).toBe(true);
+                expect(mask2.includeInBuild).toBe(false);
+                expect(mask1.measurable).toBe(false);
+                expect(mask2.measurable).toBe(true);
+
+                container1.mask = mask2;
+
+                expect(mask1.includeInBuild).toBe(true);
+                expect(mask1.measurable).toBe(false);
+                expect(mask2.includeInBuild).toBe(true);
+                expect(mask2.measurable).toBe(false);
+
+                container2.mask = mask2;
+
+                expect(mask1.includeInBuild).toBe(false);
+                expect(mask1.measurable).toBe(true);
+                expect(mask2.includeInBuild).toBe(true);
+                expect(mask2.measurable).toBe(false);
+
+                container1.mask = null;
+
+                expect(mask1.includeInBuild).toBe(false);
+                expect(mask1.measurable).toBe(true);
+                expect(mask2.includeInBuild).toBe(true);
+                expect(mask2.measurable).toBe(false);
+
+                container2.mask = null;
+
+                expect(mask1.includeInBuild).toBe(false);
+                expect(mask1.measurable).toBe(true);
+                expect(mask2.includeInBuild).toBe(false);
+                expect(mask2.measurable).toBe(true);
+
+                container1.mask = mask1;
+
+                expect(mask1.includeInBuild).toBe(true);
+                expect(mask1.measurable).toBe(false);
+
+                container1.destroy();
+
+                expect(mask1.includeInBuild).toBe(false);
+                expect(mask1.measurable).toBe(true);
+            }
+        );
+    });
+
+    describe('alpha', () =>
+    {
+        it('should traverse alpha', () =>
+        {
+            const rootContainer = new Container({ layer: true });
+            const parent = new Container();
+            const child = new Container();
+
+            rootContainer.addChild(parent);
+            parent.addChild(child);
+
+            parent.alpha = 0.5;
+
+            updateLayerGroupTransforms(rootContainer.layerGroup, true);
+
+            const alpha = ((child.layerColor >> 24) & 0xFF) / 255;
+
+            expect(alpha).toBeCloseTo(0.5, 2);
```

Comment:
we no longer need to do this

we can just directly test the alpha

Reply:
I simplified the test to check the `child.layerAlpha` since its really just testing that parent alpha values are cascaded.